src/main/java/com/snowplowanalytics/snowplow/tracker/Tracker.java

```diff
@@ -190,7 +195,98 @@ public TrackerParameters getParameters() {
+        // a list because Ecommerce events become multiple payloads
+        List<Event> eventList = new ArrayList<>();
+
+        final Class<?> eventClass = event.getClass();
+        if (eventClass.equals(Unstructured.class)) {
+            // Need to set the Base64 rule for Unstructured events
+            final Unstructured unstructured = (Unstructured) event;
+            unstructured.setBase64Encode(this.parameters.getBase64Encoded());
+            eventList.add(unstructured);
+
+        } else if (eventClass.equals(EcommerceTransaction.class)) {
+
+            final EcommerceTransaction ecommerceTransaction = (EcommerceTransaction) event;
+            eventList.add(ecommerceTransaction);
+
+            // Track each item individually
+            for (final EcommerceTransactionItem item : ecommerceTransaction.getItems()) {
+                item.setDeviceCreatedTimestamp(ecommerceTransaction.getDeviceCreatedTimestamp());
+                eventList.add(item);
+            }
+        } else if (eventClass.equals(Timing.class) || eventClass.equals(ScreenView.class)) {
+            // These are wrapper classes for Unstructured events; need to create
+            // Unstructured events from them and resend.
+            final Unstructured unstructured = Unstructured.builder()
+                    .eventData((SelfDescribingJson) event.getPayload())
+                    .customContext(event.getContext())
+                    .deviceCreatedTimestamp(event.getDeviceCreatedTimestamp())
+                    .trueTimestamp(event.getTrueTimestamp())
+                    .eventId(event.getEventId())
+                    .subject(event.getSubject())
+                    .build();
+
+            unstructured.setBase64Encode(this.parameters.getBase64Encoded());
+            eventList.add(unstructured);
+        } else {
+            eventList.add(event);
+        }
+        return eventList;
+    }
+
+    private void addTrackerParameters(TrackerPayload payload) {
+        payload.add(Parameter.PLATFORM, this.parameters.getPlatform().toString());
+        payload.add(Parameter.APP_ID, this.parameters.getAppId());
+        payload.add(Parameter.NAMESPACE, this.parameters.getNamespace());
+        payload.add(Parameter.TRACKER_VERSION, this.parameters.getTrackerVersion());
+    }
+
+    private void addContext(Event event, TrackerPayload payload) {
+        List<SelfDescribingJson> entities = event.getContext();
+
+        // Build the final context and add it to the payload
+        if (entities != null && entities.size() > 0) {
+            SelfDescribingJson envelope = getFinalContext(entities);
+            payload.addMap(envelope.getMap(), this.parameters.getBase64Encoded(), Parameter.CONTEXT_ENCODED, Parameter.CONTEXT);
+        }
+    }
+
+    /**
+     * Builds the final event context.
+     *
+     * @param entities the base event context
+     * @return the final event context json with many entities inside
+     */
```

Comment:
Opinion sharing:
I don't see much value in overzealous javadoc on private methods (I know you just copied/pasted this code).
I feel that in private methods we can remove these when the method signature is already self-explanatory, or just leaving a simple comment like "Builds the final event context." without the other rows.
Not something to change, btw.